Gravatar/MruImageCache.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+
+namespace Gravatar
+{
+    /// <summary>
+    /// Caches most-recently-used images.
+    /// </summary>
+    /// <remarks>
+    /// Decorates an inner cache, delegating to it as needed.
+    /// <para />
+    /// If an image is available in memory, the inner cache can be bypassed.
+    /// </remarks>
+    public sealed class MruImageCache : IImageCache
+    {
+        private readonly int _cleanAtSize;
+        private readonly int _cleanToSize;
+
+        private readonly ConcurrentDictionary<string, Entry> _entryByFileName = new ConcurrentDictionary<string, Entry>();
+        private readonly IImageCache _inner;
+
+        public MruImageCache([NotNull] IImageCache inner, int cleanAtSize = 30, int cleanToSize = 25)
```

Comment:
:bulb: This can be simplified by making the size limit a private constant of the type. (The separation into `cleanAtSize` and `cleanToSize` will be eliminated by the switch to a linked list.)

Reply:
It was a constant, until I wanted to test the type with smaller values.